OneCommunityGlobal/HighestGoodNetworkApp
# Lucas user hours set to zero

This PR solves (PRIORITY LOW) Jae: Special Situation - User Hours Set to Zero

### PLEASE USE THE PROPER BACKEND ([#298](https://github.com/OneCommunityGlobal/HGNRest/pull/298)) SO THE PR CAN WORK PROPERLY

<hr>

### Main changes

Now, Mentor users and users with zero hours are not displayed on the leaderboard and do not add to the HGN totals. Mentor users can only be seen by themselves and Owners, Administrators and Core Teams on the Leaderboard. Also, Mentor users and users with zero hours can see the members on their respective teams.

<hr>

### How to test

- Checkout the current branch on both frontend and backend provided on Slack and on this PR
- npm install (on the backend)
- npm run build (on the backend)
- npm run start (on the backend)
- npm install (on the frontend)
- npm start:local (on the frontend)
- Login as various roles and check the users that appear on the Leaderboard (Mentors and zero hour users should see themselves and team members, Owners, Administrators and Core Team members should be able to see all users, including zero hour users and Mentor users. Other role's users should not see Mentor users and zero hour users)
- Change some user's hours to zero and role to Mentor and check if the the count on HGN Totals changes.
